From 032f0a503c5b6858e894defd0f8239ebecd9d26f Mon Sep 17 00:00:00 2001
From: Vladimir Ermakov <vooon341@gmail.com>
Date: Sat, 20 Dec 2014 08:05:13 +0300
Subject: [PATCH 3/5] ipnc: Add eth= cmdline support

Original camera U-boot pass MAC address in cmdline option.
I decided to keep that option.

Signed-off-by: Vladimir Ermakov <vooon341@gmail.com>
---
 arch/arm/mach-davinci/board-ts38f2-ipnc.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-davinci/board-ts38f2-ipnc.c b/arch/arm/mach-davinci/board-ts38f2-ipnc.c
index ea8ed58..80a0dda 100644
--- a/arch/arm/mach-davinci/board-ts38f2-ipnc.c
+++ b/arch/arm/mach-davinci/board-ts38f2-ipnc.c
@@ -49,7 +49,6 @@
 
 
 /* TODO:
- * - parse cmdargs eth=mac value
  * - Composite video out
  * - Aptina MT9P031 sensor description
  * - IR CUT driver (separate file)
@@ -297,6 +296,18 @@ static void __init ipnc_init_status_led(void)
 	davinci_cfg_reg(DM365_GPIO33);
 }
 
+/* EMAC MAC adress passed via `mac=` cmdline argument */
+static int __init ipnc_eth_mac_setup(char *str)
+{
+	struct davinci_soc_info *soc_info = &davinci_soc_info;
+
+	if (mac_pton(str, soc_info->emac_pdata->mac_addr))
+		pr_info("EMAC: Use MAC from eth=%pM", soc_info->emac_pdata->mac_addr);
+
+	return 1;
+}
+__setup("eth=", ipnc_eth_mac_setup);
+
 /*
  * Board init
  */
-- 
2.1.0

